scala.html.markdown

```diff
@@ -253,16 +253,20 @@ weirdSum(2, 4) // => 16
 // def that surrounds it.
 // WARNING: Using return in Scala is error-prone and should be avoided.
 // It has no effect on anonymous functions. For example:
-def foo(x: Int): Int = {
-  val anonFunc: Int => Int = { z =>
+def addTenButMaybeTwelve(x: Int): Int = {
+  val anonMaybeAddTwo: Int => Int = { z =>
     if (z > 5)
-      return z // This line makes z the return value of foo!
+      return z // This line makes z the return value of addTenButMaybeTwelve!
     else
       z + 2    // This line is the return value of anonFunc
   }
-  anonFunc(x)  // This line is the return value of foo
+  anonMaybeAddTwo(x) + 10  // This line is the return value of foo
```

Comment:
Well it is the return value usually, correct? It's just the explicit `return` statement above that short-circuits the function and skips the default return value. Adding `return` for the else would miss the point of the example again.

Would it be clearer to add "This line is the default return value for addTenButMaybeTwelve"? Or we could just leave the comment off and be a little less explicit, but not potentially add confusion.